packages/camera/camera_windows/lib/camera_windows.dart

```diff
@@ -245,52 +238,24 @@ class CameraWindows extends CameraPlatform {
   @override
   Stream<CameraImageData> onStreamedFrameAvailable(int cameraId,
       {CameraImageStreamOptions? options}) {
-    _installStreamController(
-        onListen: () => _onFrameStreamListen(cameraId),
-        onCancel: () => _onFrameStreamCancel(cameraId));
-    return _frameStreamController!.stream;
-  }
-
-  StreamController<CameraImageData> _installStreamController(
-      {void Function()? onListen, void Function()? onCancel}) {
-    _frameStreamController = StreamController<CameraImageData>(
-      onListen: onListen ?? () {},
-      onPause: _onFrameStreamPauseResume,
-      onResume: _onFrameStreamPauseResume,
-      onCancel: onCancel ?? () {},
-    );
-    return _frameStreamController!;
-  }
-
-  void _onFrameStreamListen(int cameraId) {
-    _startPlatformStream(cameraId);
-  }
-
-  Future<void> _startPlatformStream(int cameraId) async {
-    await _hostApi.startImageStream(cameraId);
-    _startStreamListener(cameraId);
-  }
-
-  void _startStreamListener(int cameraId) {
-    final eventChannelName =
-        'plugins.flutter.io/camera_windows/imageStream/$cameraId';
-    final EventChannel cameraEventChannel = EventChannel(eventChannelName);
-    _platformImageStreamSubscription =
-        cameraEventChannel.receiveBroadcastStream().listen((dynamic imageData) {
-      _frameStreamController!
-          .add(cameraImageFromPlatformData(imageData as Map<dynamic, dynamic>));
-    });
-  }
-
-  FutureOr<void> _onFrameStreamCancel(int cameraId) async {
-    await _stopPlatformStream(cameraId);
-  }
-
-  Future<void> _stopPlatformStream(int cameraId) async {
-    await _hostApi.stopImageStream(cameraId);
-    await _platformImageStreamSubscription?.cancel();
-    _platformImageStreamSubscription = null;
-    _frameStreamController = null;
+    late StreamController<CameraImageData> controller;
+
+    controller = StreamController<CameraImageData>(
+        onListen: () async {
+          final eventChannelName = await _hostApi.startImageStream(cameraId);
+          final EventChannel imageStreamChannel =
+              EventChannel(eventChannelName);
+          imageStreamChannel.receiveBroadcastStream().listen((dynamic image) =>
```

Comment:
This creates subscription.
should it be cancelled in onCancel as well.

having 
late StreamSubscription<dynamic> subscription;

save the subscription:
subscription = imageStreamChannel.receiveBroadcastStream().listen ...

cancel the subscription:

  onCancel: () async {
    await subscription.cancel();
    await _hostApi.stopImageStream(cameraId);
  },
  
  ?
  Not sure if this is needed though.